src/java.base/share/classes/java/time/ZoneId.java

```diff
@@ -162,11 +162,9 @@
  * queried, but not modified, on a Java Runtime with incomplete time-zone information.
  * <p>
  * This is a <a href="{@docRoot}/java.base/java/lang/doc-files/ValueBased.html">value-based</a>
- * class; programmers should treat instances that are
- * {@linkplain #equals(Object) equal} as interchangeable and should not
- * use instances for synchronization, or unpredictable behavior may
- * occur. For example, in a future release, synchronization may fail.
- * The {@code equals} method should be used for comparisons.
+ * class; programmers should treat instances that are {@linkplain #equals(Object) equal}
+ * as interchangeable and should not use instances for synchronization, mutexes, or
+ * with {@linkplain java.lang.ref.Reference object references}.
```

Comment:
Should we add the preview warning with rewords, like
```
When preview features are enabled, all implementations of {@code ZoneId} are {@linkplain Class#isValue value class}. ...
```

Reply:
There's more to do before ZoneId can be a Valhalla value class.
Its subtype ZoneOffset has a non-final field that caches ZoneRules. So ZoneOffset can't be a value class.